test/e2e/storage/testsuites/snapshottable_stress.go

```diff
@@ -0,0 +1,233 @@
+}
+
+func (t *snapshottableStressTestSuite) SkipRedundantSuite(driver TestDriver, pattern testpatterns.TestPattern) {
+}
+
+func (t *snapshottableStressTestSuite) DefineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	var (
+		driverInfo          *DriverInfo
+		snapshottableDriver SnapshottableTestDriver
+		cs                  clientset.Interface
+		stressTest          *snapshottableStressTest
+	)
+
+	// Check preconditions before setting up namespace via framework below.
+	ginkgo.BeforeEach(func() {
+		driverInfo = driver.GetDriverInfo()
+		if driverInfo.StressTestOptions == nil {
+			e2eskipper.Skipf("Driver %s doesn't specify stress test options -- skipping", driverInfo.Name)
+		}
+
+		// Because we're initializing snapshottableDriver, both vars must exist.
+		ok := false
+
+		snapshottableDriver, ok = driver.(SnapshottableTestDriver)
+		if !driverInfo.Capabilities[CapSnapshotDataSource] || !ok {
+			e2eskipper.Skipf("Driver %q doesn't implement SnapshottableTestDriver - skipping", driverInfo.Name)
+		}
+
+		_, ok = driver.(DynamicPVTestDriver)
+		if !ok {
+			e2eskipper.Skipf("Driver %s doesn't implement DynamicPVTestDriver -- skipping", driverInfo.Name)
+		}
+	})
+
+	// This intentionally comes after checking the preconditions because it
+	// registers its own BeforeEach which creates the namespace. Beware that it
+	// also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewDefaultFramework("snapshottable-stress")
+
+	init := func() {
+		cs = f.ClientSet
+		config, driverCleanup := driver.PrepareTest(f)
+		ctx, cancel := context.WithCancel(context.Background())
+
+		stressTest = &snapshottableStressTest{
+			config:        config,
+			driverCleanup: driverCleanup,
+			volumes:       []*VolumeResource{},
+			snapshots:     []*SnapshotResource{},
+			pods:          []*v1.Pod{},
+			testOptions:   *driverInfo.StressTestOptions,
+			ctx:           ctx,
+			cancel:        cancel,
+		}
+	}
+
+	createPodsAndVolumes := func() {
+		for i := 0; i < stressTest.testOptions.NumPods; i++ {
+			framework.Logf("Creating resources for pod %v/%v", i, stressTest.testOptions.NumPods-1)
+
+			volume := CreateVolumeResource(driver, stressTest.config, pattern, t.GetTestSuiteInfo().SupportedSizeRange)
+			stressTest.volumes = append(stressTest.volumes, volume)
+
+			podConfig := e2epod.Config{
+				NS:           f.Namespace.Name,
+				PVCs:         []*v1.PersistentVolumeClaim{volume.Pvc},
+				SeLinuxLabel: e2epv.SELinuxLabel,
+			}
+			pod, err := e2epod.MakeSecPod(&podConfig)
+			framework.ExpectNoError(err)
+			stressTest.pods = append(stressTest.pods, pod)
+
+			if _, err := cs.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{}); err != nil {
+				stressTest.cancel()
+				framework.Failf("Failed to create pod-%v [%+v]. Error: %v", i, pod, err)
+			}
+
+			if err := e2epod.WaitForPodRunningInNamespace(cs, pod); err != nil {
```

Comment:
This may take some time to serially wait for each pod to come up. Can we create all the pods at once, and then have a second for loop to wait for each pod to come up?

Reply:
Similar thing for delete

Reply:
Force pushed an update for this, the test completes much faster now